src/ls/definition.cr

```diff
@@ -0,0 +1,166 @@
+module Mint
+  module LS
+    class StackReader
+      def initialize(@stack : Array(Ast::Node))
+        @index = 0
+      end
+
+      def find_next(klass : N.class) : N | Nil forall N
+        node = @stack[@index]?.try &.as?(N)
+        @index += 1
+        node
+      end
+
+      def find_anywhere(klass : N.class) : N | Nil forall N
+        @stack[@index..].each do
+          node = find_next(klass)
+          return node if node
+        end
+      end
+    end
+
+    # This is the class that handles the "textDocument/definition" request.
+    class Definition < LSP::RequestMessage
+      property params : LSP::TextDocumentPositionParams
+
+      def execute(server) : LSP::LocationLink | LSP::Location | Nil
+        uri =
+          URI.parse(params.text_document.uri)
+
+        workspace =
+          Workspace[uri.path.to_s]
+
+        unless workspace.error
```

Comment:
Do we need to return something in case we have an error?

Reply:
I think we should probably be returning a [`ResponseError`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#responseError) for these errors, but I think that would require a bigger refactoring of the `LSP::Server` :sweat_smile: 

Without that, I think `null` is probably the most appropriate out of the definition responses (i.e.  ```Location | Location[] | LocationLink[] | null```)